db/migrate/20211116041805_add_search_status_and_role_preferences_to_developers.rb

```diff
@@ -0,0 +1,9 @@
+class AddSearchStatusAndRolePreferencesToDevelopers < ActiveRecord::Migration[7.0]
+  def change
+    add_column :developers, :search_status, :integer
+
+    add_column :developers, :part_time_contract, :boolean
+    add_column :developers, :full_time_contract, :boolean
+    add_column :developers, :full_time_employment, :boolean
```

Comment:
To represent some checkboxes, I recently used an enum on the model and stored the values in one string array column:

```suggestion
    add_column :developers, :role_types, :string, array: true
```

The model had a scope similar to this: 

```ruby
  scope :with_role, ->(value) do
    where arel_table[:role_types].array_contains value
  end
```

The scope is reliant on a method I have in `lib/rails_ext/arel/predications.rb`:

```ruby
module Arel
  module Predications
    ...
    def array_contains(*rhs)
      Nodes::InfixOperation.new "@>", self, Nodes.build_quoted(to_array_literal(rhs))
    end
    ...
  end
end
```

Reply:
Ah, very cool! I've shied away from array columns in the past because if/when you change the values the migrations can be a pain. I can see that helper and scope being super useful, though! I'll keep this in my back pocket if we ever migrate to an array.

I'm leaning more towards a `has_one` first-class model now.

Reply:
> if/when you change the values the migrations can be a pain

Yes, I know what you mean. I stuck to ruby objects for the migration I implemented recently e.g.

```ruby
    Developer.find_each do |developer|
      developer.role_types << :part_time_contract
      developer.role_types << :full_time_employment unless some condition
      developer.save(validate: false)
    end
```